platform/mikro-e/dev/proximity-click.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2016, Imagination Technologies Limited and/or its
+ * affiliated group companies.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the author may not be used to endorse or promote
+ *    products derived from this software without specific prior
+ *    written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include <p32xxxx.h>
+#include <pic32_i2c.h>
+#include "proximity-click.h"
+
+#define I2C_FREQUENCY		          100000
+#define SALE_ADDR_WRITE               0X26
+#define SALE_ADDR_READ                0X27
+/* VCNL4010 REGISTERS */
+#define COMMAND_REG                   0X80
+#define PROXIMITY_RATE_REG            0X82
+#define IRLED_CUURENT_REG             0X83
+#define PROXIMITY_DATA_UPPER_REG      0X87
+#define PROXIMITY_DATA_LOWER_REG      0X88
+#define HIGH_THRESHOLD_UPPER_REG      0X8C
+#define HIGH_THRESHOLD_LOWER_REG      0X8D
+#define INTERRUPT_CONTROL_REG         0X89
+#define INTERRUPT_STATUS_REG          0X8E
+/* reg config value */
+#define PROXIMITY_SELFTIME_EN         0X03
+#define PROXIMITY_RATE_4              0X01
+#define IRLED_CURRENT_100MA           0X10
+#define THRESHOLD_UPPER_BYTE          0X08
+#define THRESHOLD_LOWER_BYTE          0XA0
+#define TH_INT_EN_TWO_COUNT           0X22
```

Comment:
TH_INT_EN_TWO_COUNT can be defined as INTERRUPT_REG_SETTING and in comment can add the details of settings.